### PR TITLE
Entity Bringdown volume 2

### PR DESCRIPTION
**Pull request by TimGoll**
Brought more entities from base TTT into TTT2. Modified the code a bit so that it adheres to our coding style, but nothing else was changed.

- targetID integration was improved
- fixed targetID hints (legacy TTT) only working for players and corpses
- updated targetId for visualizer so that it shows that only public policing roles can pick it up
- added missing translatable strings
- removed some leftover semicolons

### CODE REVIEW

File: gamemodes/terrortown/entities/weapons/weapon_ttt_push.lua [hunk 1]
```diff
@@ -276,17 +279,17 @@ if CLIENT then
 			surface.DrawOutlinedRect(x - w / 2, y - h, w, h, 1)
 
 			if LocalPlayer():IsTraitor() then
-			   surface.SetDrawColor(255, 0, 0, 155)
+			 surface.SetDrawColor(255, 0, 0, 155)
```
Comment (saibotk) [inline gamemodes/terrortown/entities/weapons/weapon_ttt_push.lua:282]:
```suggestion
				surface.SetDrawColor(255, 0, 0, 155)
```

**Review by saibotk — CHANGES_REQUESTED** [review]

File: gamemodes/terrortown/entities/entities/ttt_cse_proj.lua [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+		self.ScanSound:Stop()
+	end
+end
+
+---
+-- @param Entity activator
+-- @realm shared
+function ENT:UseOverride(activator)
+	if IsValid(activator) and activator:IsPlayer() then
+		if activator:IsActiveDetective() and activator:CanCarryType(WEAPON_EQUIP) then
```
[the remark this answers is not in the document]
Comment (TimGoll) [reply gamemodes/terrortown/entities/entities/ttt_cse_proj.lua:137]: probably a good idea. Only public policing roles should be able to pick it up

File: gamemodes/terrortown/entities/entities/ttt_knife_proj.lua [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+		wep.IsDropped = true
+
+		local prints = self.fingerprints or {}
+
+		self:Remove()
+
+		wep:Spawn()
+		wep.fingerprints = wep.fingerprints or {}
+
+		wep.fingerprints[#wep.fingerprints + 1] = prints
```
[the remark this answers is not in the document]
Comment (TimGoll) [reply gamemodes/terrortown/entities/entities/ttt_knife_proj.lua:186]: Honestly no idea. I took this as is from TTT. If you think this is a problem, I can investigate it

[the remark this answers is not in the document]
Comment (saibotk) [reply gamemodes/terrortown/entities/entities/ttt_knife_proj.lua:186]: Would be good, this seems like an error and we should check that before we include this into our codebase.

[the remark this answers is not in the document]
Comment (EntranceJew) [reply gamemodes/terrortown/entities/entities/ttt_knife_proj.lua:186]: I encountered this before while working on a tweaked version of a knife, I'm fairly confident it is an error, as the code should more than likely read:
```lua
wep.IsDropped = true
wep.fingerprints = table.Copy( self.fingerprints or {} )

self:Remove()

wep:Spawn()
```

I think they were confused by the prospects of `self:Remove()` destroying `self.fingerprints` instantly, and were distracted by copying by reference the table and potentially not its data and forgot to iterate it. Or they were going to add it to the player stack but didn't use the same pattern as the health station. Pretty much every item in a fingerprints table is meant to be a Player, so, this should change.

**Review by Histalek — APPROVED** [review]

File: gamemodes/terrortown/entities/entities/ttt_knife_proj.lua (hunk 3 above)
[the remark this answers is not in the document]
Comment (TimGoll) [reply gamemodes/terrortown/entities/entities/ttt_knife_proj.lua]: done

**Review by EntranceJew — APPROVED** [review]
i pulled down the files and compared the new ones to `Facepunch/garrysmod`:

things I assume are intentional:
 - leaving `ttt_c4/cl_init.lua` (C4 / Bomb UI) upstream still
 - all the grenades remaining upstream (grenade rework pending)
 - `ttt_weapon_check` remaining upstream
 - `ttt_carry_handler` (not utilized by `weapon_zm_carry`) 

weapons that still only exist on the upstream:
 - `weapon_ttt_c4`
 - `weapon_ttt_defuser`
 - `weapon_ttt_flaregun`
 - `weapon_ttt_health_station`
 - `weapon_ttt_radio`
 - `weapon_ttt_sipistol`
 - `weapon_ttt_stungun`
 - `weapon_ttt_unarmed`

File: gamemodes/terrortown/entities/entities/ttt_cse_proj.lua [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+function ENT:GetNearbyCorpses()
+	local pos = self:GetPos()
+
+	local near = ents.FindInSphere(pos, self.Range)
+	if not near then return end
+
+	local near_corpses = {}
+
+	for i = 1, #near do
+		local ent = near[i]
```
[the remark this answers is not in the document]
Comment (TimGoll) [reply gamemodes/terrortown/entities/entities/ttt_cse_proj.lua:49]: That's true. It is slightly faster to define the variable outside the loop and just update the value on each iteration. In our codebase we never use that approach though. Therefore I decided to change that here. To keep it consistent.

**Comment by TimGoll** [comment]
About the missing entities: They could be part of volume 3. Imho it is better to keep them seperate to make the review easier

**Comment by EntranceJew** [comment]
> About the missing entities: They could be part of volume 3. Imho it is better to keep them seperate to make the review easier

Absolutely, I was just making note since I had the files pulled down for comparison.

**Review by saibotk — APPROVED** [review]

**Pull request merged**